### PR TITLE
Keep filename on import

### PR DESCRIPTION
**Pull request by jdcook**
Suggested code to update filename after import. Without this, the files get named "Untitled.00X"

### CODE REVIEW

**Comment by p2or** [comment]
Hi @jdcook,

thanks for your PR. As much as I appreciate your effort, I personally see several problems. However, we can discuss them with pleasure:

1. Just to make sure: Could you please elaborate on why it is useful to remove the digits?
1. What if the user would like to keep the digits? Don't you think it would be good to offer the user an option to turn this behavior on or off?

**Technical details**

1. Consider that as is, the PR removes the entire part after the last dot appearing in the name (including the dot of course) and *not only* the digits. So if there is an object with a regular dot in its name, like `jon.doe`, the last part `.doe` will be removed as well right? Is this intended?
1. As of Python 3.4 there is a new `pathlib` module and my idea was actually to get rid of the `os` module used in my original version. I don't think it is necessary to import an entire library for only one function call. Or is there no substitute? 
   - https://docs.python.org/3/library/pathlib.html
   - https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/
1. Why the for-loop within the main loop? As is, the PR updates the filename of all objects in selection multiple times (after one object gets imported). Isn't `context.object.name = "new-name"` enough in this case?
1. Variable names should be `lower_case` according to [`pep8`](https://peps.python.org/pep-0008/#descriptive-naming-styles)

Cheers,
Christian

File: io_batch_import_objs.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
 
 
 import bpy
+import os
```
Comment (p2or) [inline io_batch_import_objs.py:34]: As of Python 3.4 there is a new `pathlib` module and my idea was actually to get rid of the `os` module used in my original version. I don't think it is necessary to import an entire library for only one function call. Or is there no substitute?

**Review by p2or — COMMENTED** [review]
Code review:

File: io_batch_import_objs.py [hunk 2]
```diff
@@ -118,6 +118,10 @@ def execute(self, context):
                                 clamp_size = self.clamp_size_setting,
                                 validate_meshes = self.validate_setting
                                 )
+                # assign filename (without extension) to the imported object
```
Comment (p2or) [inline io_batch_import_objs.py:122]: Why the for-loop within the main loop? As is, the PR updates the filename of all objects in selection multiple times (after one object gets imported). Isn't `context.object.name = "new-name"` enough in this case?

**Comment by jdcook** [comment]
Hi @p2or, thanks for the detailed review, the purpose of the code is to keep the filename as the imported object's name, not to remove the digits at the end. So "character1.obj", "character2.obj" would be imported as the "character1", "character2" objects in Blender, instead of "Untitled.001", "Untitled.002". The split is removing the file extension.

selected_objects gets updated to the newly created object after the import. I can see if there's another way to select the just-imported object if you'd like though, I'm new to Blender scripts so I didn't research that thoroughly. Happy to update the naming convention and library used if there's an equivalent function in pathlib.

Are you interested in an update or does closing the PR mean you aren't interested in the feature?

**Comment by p2or** [comment]
Hi @jdcook,

> So "character1.obj", "character2.obj" would be imported as the "character1", "character2" objects in Blender, instead of "Untitled.001", "Untitled.002". The split is removing the file extension.

thanks for your explanation. If this is the case, I guess your `.obj` files do not conform to the [official specifications](https://en.wikipedia.org/wiki/Wavefront_.obj_file#File_format) and `o [object name]` tag is missing from the definition. How the obj file should look like:

```python
...
mtllib skin.mtl
o Character1
v 1.000000 1.000000 -1.000000
...
```

If the `o` tag is specified, the file name will be "kept" by default. Example of two objects named `Cube` and `Cube.001` exported from Blender and imported back using the addon. As you can see the name is properly assigned to each object:

Screenshot 2022-08-28 at 10 11 44

Both files to test: [objs.zip](https://github.com/p2or/blender-batch-import-wavefront-obj/files/9439313/objs.zip)

If that's your issue, I'd suggest to file a bug report to whatever DCC you are using. The only good option I see right now is to parse each file for the object tag before importing. For a quick and dirty "fix", you could also add a simple `if` statement at the end of the `for`-loop in order to import your corrupted files and rename the objects according to their respective file names:

```python
    ...

    if context.object.name.startswith("Untitled"):
        context.object.name = fp.stem

return {'FINISHED'}

```

Does this help? Otherwise please share one of your `obj` files to test.

Cheers,
Christian

**Comment by jdcook** [comment]
Oh dang you're right, it's actually a problem with MagicaVoxel's obj exports. Here's an example header from their file:

```
# MagicaVoxel @ Ephtracy

# group
o

# material

...
```

Adding the name after the "o" tag makes it import correctly. Thanks for the explanation @p2or, learned a lot from it, really appreciate the work you put into the addon and the maintenance you do to keep up with Blender versions! It saves me a bunch of time when importing animation frames exported from MagicaVoxel at once.

**Comment by p2or** [comment]
Cool, glad I could help. Cheers

**Comment by jdcook** [comment]
@p2or FYI if the info helps, today I realized that the normal .obj importer did import my models with the name intact from MagicaVoxel, and looking into the code for that, I realized it also grabs the filename for the object:

https://github.com/blender/blender-addons/blob/master/io_scene_obj/import_obj.py#L477
```
    filename = os.path.splitext((os.path.basename(filepath)))[0]

    if not SPLIT_OB_OR_GROUP or not faces:
        use_verts_nor = any(f[1] for f in faces)
        use_verts_tex = any(f[2] for f in faces)
        # use the filename for the object name since we aren't chopping up the mesh.
        return [(verts_loc, faces, unique_materials, filename, use_verts_nor, use_verts_tex)]
```

So something to consider for consistency, that's why it confused me why my objects weren't named when switching to a batch importer.

**Pull request closed without merging**